### ISSUE TITLE
Undergraduate Machine Learning 30

### ISSUE THREAD

**Issue by Yajieniu** [Yajieniu/CMS-Progress-Mark#14]
## Deep Learning

1. In the real world, there is a lot of unsupervised learning.
2. PCA: given a matrix X, uses two matrixes B and C to represent X (reconstruction).
3. Unsupervised Feature Learning/Self-taught learning
    1) Encoding: project a very large image to a low dimensional space. It allows you to get a low dimensional representation of the image.
    2) Transfer learning, also called self-taught learning, means you learn the theory somewhere, and you are able to transform it into a different world. Thus we could recognize the high-level features after we learned them in supervised or unsupervised ways.
    3) Applying SVD on the large data matrix is too expensive, so instead, we compute the derivative (gradient descent).
4. Multi-layers -- greedy layer-wise training: unsupervised pre-training + fine-tuning.
    1) Learn the hidden layer (the first layer).
    2) Get a second layer of features, which is trying to predict itself and so on. 
    3) Learn the final classifier.
    4) Combine everything together and we get a deep neural network.
    5) *Back propagation is very expensive. Use gradient descent.
5.  Google autoencoder.
    1) Pooling: pick the average of several neurons in order to compress images.
